packages/web3-core/src/types.ts

```diff
@@ -40,6 +43,16 @@ export type Web3BaseProviderConstructor = new <API extends Web3APISpec>(
 	net?: Socket,
 ) => Web3BaseProvider<API>;
 
+// TODO: When we have `web3-types` package we can share TransactionType
+export type TransactionTypeParser = (transaction: Record<string, unknown>) => HexString | undefined;
+
+// TODO: When we have `web3-types` package we can share TransactionType
+export type TransactionBuilder = (options: {
```

Comment:
Any reason to pass parameters as an object here?

Reply:
Object parameters are easy to extend in future without making breaking changes. It's also easy to read when used.